packages/ember-engines/addon/components/link-to-external.js

```diff
@@ -1,53 +1,933 @@
+    }),
+    _isActive(routerState /* RouterState */) {
+        if (this.loading) {
+            return false;
+        }
+        let currentWhen = this['current-when'];
+        if (typeof currentWhen === 'boolean') {
+            return currentWhen;
+        }
+        let { _models: models, _routing: routing } = this;
+        if (typeof currentWhen === 'string') {
+            return currentWhen
+                .split(' ')
+                .some((route) => routing.isActiveForRoute(models, undefined, this._namespaceRoute(route), routerState));
+        }
+        else {
+            return routing.isActiveForRoute(models, this._query, this._route, routerState);
+        }
+    },
+    transitioningIn: computed('_active', 'willBeActive', function computeLinkToComponentTransitioningIn() {
+        if (this.willBeActive === true && !this._active) {
+            return 'ember-transitioning-in';
+        }
+        else {
+            return false;
+        }
+    }),
+    transitioningOut: computed('_active', 'willBeActive', function computeLinkToComponentTransitioningOut() {
+        if (this.willBeActive === false && this._active) {
+            return 'ember-transitioning-out';
+        }
+        else {
+            return false;
+        }
+    }),
+    _namespaceRoute(route) {
+        let { _engineMountPoint: mountPoint } = this;
+        if (mountPoint === undefined) {
+            return route;
+        }
+        else if (route === 'application') {
+            return mountPoint;
+        }
+        else {
+            return `${mountPoint}.${route}`;
+        }
+    },
+    /**
+      Event handler that invokes the link, activating the associated route.
+  
+      @method _invoke
+      @param {Event} event
+      @private
+    */
+    _invoke(event) {
+        if (!isSimpleClick(event)) {
+            return true;
+        }
+        let { bubbles, preventDefault } = this;
+        let target = this.element.target;
+        let isSelf = !target || target === '_self';
+        if (preventDefault !== false && isSelf) {
+            event.preventDefault();
+        }
+        if (bubbles === false) {
+            event.stopPropagation();
+        }
+        if (this._isDisabled) {
+            return false;
+        }
+        if (this.loading) {
+            // tslint:disable-next-line:max-line-length
+            warn('This link is in an inactive loading state because at least one of its models ' +
+                'currently has a null/undefined value, or the provided route name is invalid.', false, {
+                id: 'ember-glimmer.link-to.inactive-loading-state',
+            });
+            return false;
+        }
+        if (!isSelf) {
+            return false;
+        }
+        let { _route: routeName, _models: models, _query: queryParams, replace: shouldReplace, } = this;
+        let payload = {
+            queryParams,
+            routeName,
+        };
+        // flaggedInstrument(
```

Comment:
@void-mAlex if we go this route, how could we ensure this component gets updated and we do not fall out of sync with `ember-source`?

Reply:
this is not from ember source, it's the legacy-component one that will never ever be updated again

Reply:
@void-mAlex if I remember correctly, we end up using `@ember/legacy-built-in-components` because we could import from and extend the class.
And we wanted to use `LinkTo` provided by `ember-source` but it's not importable any more.

If we go this route, why don't we copy-paste `LinkTo` from `ember-source`? am I missing something?

Reply:
this way I can guarantee that the only 'breaking' change you really need is the re-export
I'm not sure exactly everything the new link to drops that the legacy one was handling for engines specifically, this way it's the least likely to cause changes in behaviour

trying to tide people over till there is a better router as I don't want to be changing this more than necessary myself

Reply:
@void-mAlex fair point!